lib/rspec/matchers.rb

```diff
@@ -698,7 +698,11 @@ def all(expected)
     #       matchers (due to how the custom matcher DSL was evaluated in 2.x,
     #       `match` could not be used there), but is no longer needed in 3.x.
     def match(expected)
-      BuiltIn::Match.new(expected)
+      if ::String === expected
+        BuiltIn::Captures.new(expected)
+      else
+        BuiltIn::Match.new(expected)
+      end
```

Comment:
Regardless, the point I was trying to make is that we should support with_captures for both regex matching against a strong and a string matching against a regex. All the specs I saw in this PR were only one way (the opposite of what I wrote here, I thought) and this logic looks like it only supports it one way.

Maybe it would be better to add a with_captures method to Match and have this always return a match instance? Besides, it seems odd to return the Captures instance if the user has not called with_captures and may not ever call that.